system/Validation/FormatRules.php

```diff
@@ -346,4 +346,33 @@ public function valid_date(?string $str = null, ?string $format = null): bool
 
         return $date !== false && $errors !== false && $errors['warning_count'] === 0 && $errors['error_count'] === 0;
     }
+
+    /**
+     * Token JWT Validator
+     *
+     * @param string $str String
+     */
+    public function valid_jwt(?string $str = null): bool
+    {
+        if (empty($str)) {
+            return false;
+        }
+
+        $segments = explode('.', $str);
+
+        if (count($segments) !== 3 || in_array('', $segments, true)) {
+            return false;
+        }
+
+        foreach ($segments as $index => $value) {
+            if ($index < 2) {
+                $encoded = rtrim(base64_encode(base64_decode($value, true)), '==');
+                if ($encoded !== $value) {
+                    return false;
+                }
+            }
+        }
+
+        return mb_strlen($segments[$index], '8bit') >= 32 && mb_strlen($segments[$index], '8bit') <= 64;
```

Comment:
Thanks for adding this! Please include test cases that fail both the minimum and maximum lengths.